app/reducers/dataset.js

```diff
@@ -105,7 +111,15 @@ export default function (state = initialState, action) {
         .setIn(['datasets', i, 'name'], name)
         .mergeIn(['datasets', i, 'versions', version, 'configuration'], fromJS(configuration))
     }
-    
+
+    case DATASET_SELECT_SCHEDULER: {
+      const {dataset: {id}, scheduler} = action
+      const i = state.get('datasets').findIndex((d) => d.get('id') === id)
+      const version = `${state.getIn(['datasets', i, 'versions']).keySeq().map(i => +i).max()}`
+
+      return state.setIn(['datasets', i, 'versions', version, 'configuration', 'options', 'scheduler'], scheduler)
+    }
```

Comment:
Please remove this action and set the scheduler on `DATASET_GENERATE_DATA_CONFIG_SUCCESS`

Reply:
On the current stage, maybe we don't need to put the scheduler information into `state.dataset`? Since this information is not used in the future?

Reply:
Could you please refresh my memory on why we would not use this info?

Reply:
Sure. In our matterMost chat, we said - `but it may be interesting to store the scheduler info in the dataset, when it is built in a scheduler` - that's where we agreed to add this info. But later on, personally speaking, I feel like we're not going to access this info anymore. As long as the build action is dispatched, where the dataset is built doesn't matter. We also discussed with Cameron in the stand-up meeting that `we don't need to know which scheduler we selected after building`. 

However, there's one case that might be confusing to the user, is that the `schedulerSelector` will always show the globally latest used scheduler, and that scheduler might be different from the scheduler used for the displayed dataset. This pheromone might be confusing for the user. If we think this issue matters, we can do something like this (for the `datasetPage` only):

![image](https://user-images.githubusercontent.com/25927551/112031469-61feff00-8b09-11eb-9bc5-2f7317743bad.png)